buildSrc/subprojects/basics/src/main/kotlin/gradlebuild/basics/tasks/ClasspathManifest.kt

```diff
@@ -52,18 +58,18 @@ abstract class ClasspathManifest : DefaultTask() {
 
     private
     fun createProperties() = Properties().also { properties ->
-        properties["runtime"] = externalDependencies.map { it.name }.joinForProperties()
-        properties["projects"] = runtimeClasspath.filter { it.isGradleModule() }.map { it.toGradleModuleName() }.joinForProperties()
+        properties["runtime"] = getRuntime().get().joinToString(",")
+        properties["projects"] = getProjects().get().joinToString(",")
         optionalProjects.get().takeIf { it.isNotEmpty() }?.let { optional ->
             properties["optional"] = optional.joinForProperties()
         }
     }
 
     private
-    fun File.isGradleModule() = name.startsWith("gradle-") || name.contains("-patched-for-gradle-")
+    fun FileSystemLocation.isGradleModule() = asFile.name.startsWith("gradle-") || name.contains("-patched-for-gradle-")
 
     private
-    fun File.toGradleModuleName() = name.substring(0, name.lastIndexOf('-'))
+    fun FileSystemLocation.toGradleModuleName() = asFile.name.substring(0, asFile.name.lastIndexOf('-'))
```

Comment:
:x: I think as this is done earlier now, this is breaking the `kotlin-compiler-embeddable-1.3.72-patched-for-gradle` name and then that file is named wrong. You may need to add special handling of this one name here.